File: python/1094.py
# -*- coding: utf-8 -*-

n = int(input())
dic = {
    "C": 0,
    "R": 0,
    "S": 0,
    "Tot": 0}

for x in range(n):
    qtd, cobaia = input().split()
    dic[cobaia]+= int(qtd)
    dic['Tot']+= int(qtd)

pC = dic['C']*100/dic['Tot']
pR = dic['R']*100/dic['Tot']
pS = dic['S']*100/dic['Tot']

print('''Total: {} cobaias
Total de coelhos: {}
Total de ratos: {}
Total de sapos: {}
Percentual de coelhos: {:.2f} %
Percentual de ratos: {:.2f} %
Percentual de sapos: {:.2f} %'''.format(
    dic['Tot'], dic['C'], dic['R'], dic['S'],
    pC, pR, pS
))

File: categorias/iniciante/c++/1094.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    int n, i, tot=0, x, totC=0, totR=0, totS=0;
    string y;
    cout.precision(2);
    cout.setf(ios::fixed);

    cin >> n;
    for (i=0; i<n; i++) {
        cin >> x >> y;
        if (y == "C") {
            totC += x;
        } else if (y == "R") {
            totR += x;
        } else if (y == "S") {
            totS += x;
        }
        tot += x;
    }

    cout << "Total: " << tot << " cobaias\n";
    cout << "Total de coelhos: " << totC << "\n";
    cout << "Total de ratos: " << totR << "\n";
    cout << "Total de sapos: " << totS << "\n";
    cout << "Percentual de coelhos: " << (totC*100.0)/tot << " %\n";
    cout << "Percentual de ratos: " << (totR*100.0)/tot << " %\n";
    cout << "Percentual de sapos: " << (totS*100.0)/tot << " %\n";
    return 0;
}